src/main/java/be/seeseemelk/mockbukkit/WorldMock.java

```diff
@@ -268,8 +269,17 @@ public boolean setSpawnLocation(int x, int y, int z)
 	@Override
 	public ChunkMock getChunkAt(int x, int z)
 	{
-		ChunkMock chunk = new ChunkMock(this, x, z);
-		return chunk;
+		return getChunkAt(new ChunkCoordinate(x, z));
+	}
+
+	public ChunkMock getChunkAt(ChunkCoordinate coordinate)
+	{
+		if (!loadedChunks.containsKey(coordinate))
+		{
+			ChunkMock chunk = new ChunkMock(this, coordinate.getX(), coordinate.getZ());
+			loadedChunks.put(coordinate, chunk);
+		}
+		return loadedChunks.get(coordinate);
```

Comment:
Using `HashMap#containsKey()` like that is a bad idea.
Internally `containsKey()` does a null check on `.get()`. Since you are getting the value here anyway, this results in the same operation being run twice, causing redundancy.
Moreso, you are running `.get()` on a value you just `.put()` into the Map yourself.

This can all be heavily improved by storing it as a value and doing a manual null check:

```suggestion
		ChunkMock chunk = loadedChunks.get(coordinate);
		if (chunk == null)
		{
			chunk = new ChunkMock(this, coordinate.getX(), coordinate.getZ());
			loadedChunks.put(coordinate, chunk);
		}
		return chunk;
```